src/opentracing-c/value.h

```diff
@@ -0,0 +1,60 @@
+#ifndef OPENTRACINGC_VALUE_H
+#define OPENTRACINGC_VALUE_H
+
+#include <opentracing-c/config.h>
+
+#include <stdint.h>
+
+#include <opentracing-c/common.h>
+
+/** @file */
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * Value types.
+ */
+typedef enum opentracing_value_type {
+    opentracing_value_bool,
+    opentracing_value_double,
+    opentracing_value_int64,
+    opentracing_value_uint64,
+    opentracing_value_string,
+    opentracing_value_null
+} opentracing_value_type;
+
+/**
+ * Tagged union that can represent a number of value types.
+ */
+typedef struct opentracing_value {
+    /** Value type. */
+    opentracing_value_type type;
+    /** Value storage. */
+    union {
+        /** Storage for Boolean value. */
+        opentracing_bool bool_value;
+        /** Storage for double value. */
+        double double_value;
+        /** Storage for 64 bit signed integer value. */
+        int64_t int64_value;
+        /** Storage for 64 bit unsigned integer value. */
+        uint64_t uint64_value;
+        /** Storage for string value. */
+        char* string_value;
```

Comment:
What's the memory ownership for string_value? Is it the user's responsibility to free it after calling start_span_with_options or set_tag? If not, does that mean that string literals need to be copied before put into an opentracing_value?

Reply:
Ya that's the case as of now. I want to avoid the whole string view issue I have seen in C++. But in C, the copy might be viewed more negatively. I have been thinking about reference counted strings or using a good C library like the Redis-extracted [sds](https://github.com/antirez/sds). Not sure what is best. Any thoughts?

Reply:
Also, here's how I expect it to be implemented given my work-in-progress Jaeger client: https://github.com/isaachier/c-client/blob/master/src/jaegertracingc/tag.h#L77.

Reply:
Doesn't code like [this](https://github.com/isaachier/c-client/blob/master/src/jaegertracingc/tracer.c#L446) cause a memory leak, then?

It should be freeing the memory from the passed tags before returning, no?

Reply:
Actually my intention which I must document is that the library cannot assume any memory is being passed as a "move" argument. So it never takes ownership of arguments. They might be one exception to this but I'm pretty consistent on that. The user might be using a stack based tag here so I must copy it into heap memory. If the user is using heap memory, he or she must free it themselves after calling this function.